department-of-veterans-affairs/caseflow
# CASEFLOW-676 CAVC input fields: show error on future date selection

Resolves [CASEFLOW-676](https://vajira.max.gov/browse/CASEFLOW-676)

### Description
Adds Future Date invalidation to the Cavc Remand entry form and mandate modal.

### Acceptance Criteria
- Given a CAVC Litigation Support user is completing the remand entry form, 
  - [x] when I select a date in the future in the Court's decision date field, validation should occur disallowing submitting
  - [x] when I select a date in the future in the Court's judgement date field, validation should occur disallowing submitting
  - [x] when I select a date in the future in the Court's mandate date field, validation should occur disallowing submitting
- Given a CAVC Litigation Support user is completing the remand mandate modal,
  - [x] when I select a date in the future in the Court's judgement date field, validation should occur disallowing submitting
  - [x] when I select a date in the future in the Court's mandate date field, validation should occur disallowing submitting

### Testing Plan
1. Login as a CAVC Lit Support User and go to a Dispatched appeal
2. Click `+ Add CAVC Remand`
3. Attempt to enter a future date on each of the three date fields, confirm it displays an error
4. Select MDR and complete the form
5. Load the CAVC Appeal you just created and select `Ready for 90 day letter`
6. Attempt to enter a future date on each of the two date fields, confirm it displays an error


### User Facing Changes
 UI | Validation error
 ---|---
CAVC Remand Entry | ![CAVC Remand Entry showing future date validation error](https://user-images.githubusercontent.com/6129479/108214407-23d38180-70fe-11eb-87b5-520258db8e77.png)
CAVC Mandate Entry | ![CAVC Mandate Entry showing future date validation error](https://user-images.githubusercontent.com/6129479/108214320-09010d00-70fe-11eb-8f91-692fbc4772c8.png)



### Storybook Story
 - [x] Storybook shows new behavior
